docker-compose.yml

```diff
@@ -1,6 +1,20 @@
 version: '3.4'
 
 services:
+
+  # As docker-compose does not provide an automatic mechanism for restarting unhealthy containers
```

Comment:
What? Are you sure about this? What about this "restart: always" option mentioned in all containers? And I had the impression unhealthy containers were indeed restarted due to that option

Reply:
yeah, I'm sure about it, docker(-compose) only checks the health but does nothing in case of an unhealthy container. In fact, I think the restart option probably can be removed (should be managed via the `labels`)

Reply:
I think the `restart: always` option is meant for another scenario: I assume it controls the behavior if e.g. the host or docker-daemon is restarted, or the container really terminates for some unknown reason. However, in our case, we want the container to restart if it becomes `unhealthy` which is not exactly the same thing :-/

Reply:
Will it still restart if the service terminates, e.g., due to some internal error?

Reply:
I will leave the `restart: always` as it should cover scenarios where the container terminates (because the entrypoint process has terminated for whatever reason).
However, the autoheal sidecar will now also gracefully handle the scenario, where the container has not terminated (e.g. because the process did not cleanly terminate, e.g. because of deadlock, dangling non-daemon processes, etc).